src/libYARP_OS/src/ThreadImpl.cpp

```diff
@@ -363,9 +327,7 @@ bool ThreadImpl::isClosing()
 
 bool ThreadImpl::isRunning()
 {
-    threadMutex->wait();
     bool b = active;
-    threadMutex->post();
     return b;
```

Comment:
Can it be simply `return active`? (without defining `bool b`)

Reply:
:+1: